### PR TITLE
export CustomizableCalendarDay styles, add story

### PR DESCRIPTION
**Pull request by a15n**
Previously if you wanted to update a single CSS style from CustomizableCalendarDay you'd need to (1) make the singular change and (2) duplicate all of the original CSS. 

With this change we can import the style definitions and update the single element using es6 spread operator or [update](https://github.com/kolodny/immutability-helper).

```
// before
const newDefaultStyles = {
  color: 'blue', // what we want to change
  // what we need to keep the same
  border: `1px solid ${color.core.borderLight}`,
  background: color.background,
  hover: {
    background: color.core.borderLight,
    border: `1px double ${color.core.borderLight}`,
    color: 'inherit',
  },
}


// after
import CustomizableCalendarDay, { defaultStyles, selectedStyles } from '../src/components/CustomizableCalendarDay';

const newDefaultStyles = {
  ...defaultStyles,
  color: 'blue',
};
```

### CODE REVIEW

File: src/components/CustomizableCalendarDay.jsx [hunk 1]
```diff
@@ -291,45 +311,28 @@ class CustomizableCalendarDay extends React.Component {
...
-          modifiers.has('blocked-calendar') && blockedCalendarStyles,
-          hoveredSpan && hoveredSpanStyles,
-          modifiers.has('after-hovered-start') && afterHoveredStartStyles,
-          modifiers.has('selected-span') && selectedSpanStyles,
-          modifiers.has('last-in-range') && lastInRangeStyles,
-          selected && selectedStyles,
-          modifiers.has('selected-start') && selectedStartStyles,
-          modifiers.has('selected-end') && selectedEndStyles,
-          isOutsideRange && blockedOutOfRangeStyles,
+          getStyles(defaultStylesWithHover, isHovered),
```
Comment (a15n) [inline src/components/CustomizableCalendarDay.jsx]: couldn't think of a good intermediate name so I just execute everything inline

File: stories/DayPickerSingleDateController.js [hunk 2]
```diff
@@ -206,6 +208,39 @@ storiesOf('DayPickerSingleDateController', module)
...
+      // extend and update styles with es6 spread operators
+      defaultStyles: {
+        ...defaultStyles,
+        color: 'blue',
+        hover: {
+          ...defaultStyles.hover,
+          color: 'blue',
+        },
+      },
+      // or immutability-helper's update
```
Comment (ljharb) [inline stories/DayPickerSingleDateController.js]: i'm not sure this library needs calling out in an example

File: src/components/CustomizableCalendarDay.jsx (hunk 1 above)
Comment (majapw) [reply src/components/CustomizableCalendarDay.jsx]: That seems reasonable to me

**Comment by a15n** [comment]
Thanks for the review. Ready to merge

**Pull request merged**